storage/src/mmr/mutable.rs

```diff
@@ -237,212 +323,341 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
 
         let digests = ops
             .iter()
-            .map(|op| MutableMmr::op_digest(hasher, op))
+            .map(|op| MutableMmr::<_, E, _, _, _>::op_digest(hasher, op))
             .collect::<Vec<_>>();
 
         proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
     }
 
-    /// Consume the store and return a [StoreState] from which the current active state of the store
-    /// can be fully recovered and fully proven.
-    pub fn to_state(mut self) -> StoreState<K, V, H> {
-        self.prune_known_inactive();
+    /// Commit any pending operations to the store, ensuring they are persisted to disk &
+    /// recoverable upon return from this function.
+    pub async fn commit(&mut self, hasher: &mut H) -> Result<(), Error> {
```

Comment:
nit: I recommend adding context here (or above `init`) that the only valid state of an MMR on-disk is at the commit point (and recovery will fallback to the last one).

Reply:
added context to init.